lib/links.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'links/parser'
+
+class Links
+
+  attr_reader :url, :parser
+
+  def initialize(url)
+    @url = url
+    @parser = Parser.new(url)
+  end
+
+  def run
+    links = parser.run
+    
+    responses = Enumerator.new do |enum| 
+
+      links.each do |l|
+        response = HTTParty.get(l)
+
+        enum << {code: response.code, link: l}
```

Comment:
Seems like you want to use [`Enumerable#map`](https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-map)